static/app/views/monitors/components/checkInTimeline.tsx

```diff
@@ -57,34 +76,108 @@ function getBucketedCheckInsPosition(
   return elapsedSinceStart / msPerPixel;
 }
 
+function generateJobTickFromBucket(
+  bucket: MonitorBucketData[number],
+  options?: Partial<JobTickData>
+) {
+  const [timestamp, envMapping] = bucket;
+  return {
+    endTs: timestamp,
+    startTs: timestamp,
+    width: 1,
+    envMapping,
+    roundedLeft: false,
+    roundedRight: false,
+    ...options,
+  };
+}
+
+export function mergeBuckets(data: MonitorBucketData) {
```

Comment:
I think I would probably move this into the utils as well.

in fact, I might actually create a `utils` folder and have each function be it's own file.